component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mobileconnect/MobileConnectAuthenticator.java

```diff
@@ -223,9 +255,10 @@ private void authenticationEndpoint(AuthenticationContext context, HttpServletRe
             String url = ConfigurationFacade.getInstance().getAuthenticationEndpointRetryURL();
             try {
                 response.sendRedirect(url);
+                throw new AuthenticationFailedException("Invalid JSON object returned " +
+                        "from the Discovery Server", e);
             } catch (IOException e1) {
-                throw new AuthenticationFailedException("response redirection failed. Invalid JSON object returned " +
-                        "from the Discovery Server", e1);
+                throw new AuthenticationFailedException("response redirection failed.", e1);
```

Comment:
In general, if you have an exception message without any variable check if you could add some context info. General format of the message is to start with a capital.